dss/util/auth/authorize.py

```diff
@@ -1,28 +1,49 @@
 import typing
+import logging
 
-from dss.util.auth import authregistry
+from dss.config import Config
+from dss.error import DSSForbiddenException, DSSException
+from .authregistry import AuthRegistry
 
+logger = logging.getLogger(__name__)
 
-class Authorize(metaclass=authregistry.AuthRegistry):
-    """ abstract class for authorization classes"""
+
+class Authorize(metaclass=AuthRegistry):
+    """
+    Base class for authentication/authorization methods.
+    This class exists solely to wrap the definition of the assert security flow.
+    Subclasses must define a security_flow method.
+    All subclasses inherit from AuthRegistry and are added to a registry on definition.
+    """
     def __init__(self):
         pass
 
-    def security_flow(self, authz_methods: typing.List[str], *args, **kwargs):
+    def security_flow(self, *args, **kwargs):
         """
         This function maps out flow for a given security config
+        Be sure to call it from the derived class to ensure the JWT has groups set
         """
-
-        raise NotImplementedError()
+        self.assert_required_parameters(kwargs, ['groups', 'token'])
```

Comment:
The more I think about it, the less good of an idea it seems to put this in `Authorize`. We should place as few restrictions and expectations as possible on basic Auth systems, and this is already requiring a group and a token - which makes a "no auth" data store impossible at the get go. (BTW we should add a "no auth whatsoever" class.)

I think we should move this to the Fusillade class - let Fusillade be the one assuming that a user's JWT (if they have one) will contain one (specified in a decorator?). It seems much better an idea to let an auth system using JWTs to check JWTs for info, but leave the `Authorize` class out of it.

Reply:
i agree, this idea of JWT inspection is sort of tied up into the DCP implementation of security. 
I moved the group inspection into the cls.Fus, i think at a later time we can add in a mechanism for NoAuth, and probably some documentation around using the scripting to disable the endpoint configuration in the yml.